docs/specifications/url.rst

```diff
@@ -0,0 +1,112 @@
+URLs
+====
+
+The goal of this document is to completely specify the construction and use of the URLs by Tahoe-LAFS for service location.
+This includes, but is not limited to, the original Foolscap-based URLs.
+These are not to be confused with the URI-like capabilities Tahoe-LAFS uses to refer to stored data.
+An attempt is also made to outline the rationale for certain choices about these URLs.
+The intended audience for this document is Tahoe-LAFS maintainers and other developers interested in interoperating with Tahoe-LAFS or these URLs.
+
+Background
+----------
+
+Tahoe-LAFS first used Foolscap_ for network communication.
+Foolscap connection setup takes as an input a Foolscap URL or a *fURL*.
+A fURL includes three components:
+
+* the base32-encoded SHA1 hash of the DER form of an x509v3 certificate
+* zero or more network addresses
+* an object identifier
+
+A Foolscap client tries to connect to each network address in turn.
+If a connection is established then TLS is negotiated.
+The server is authenticated by matching its certificate against the hash in the fURL.
+A matching certificate serves as proof that the handshaking peer is the correct server.
+This serves as the process by which the client authenticates the server.
+
+The client can then exercise further Foolscap functionality using the fURL's object identifier.
+If the object identifier is an unguessable, secret string then it serves as a capability.
+This serves as the process by which the server authorizes the client.
```

Comment:
"This" -> "The unguessable identifier" (if that's indeed what you mean by "this") would make the sentence more clear. Perhaps also worth introducing "swissnum" here? (Because it's used elsewhere when talking about foolscap etc)